extensions/wikia/Email/EmailController.class.php

```diff
@@ -396,6 +412,106 @@ protected function getSocialMessages() {
+					'link' => $mobileApplicationsLinks[self::ANDROID_PLATFORM],
+					'src' => EmailMobileBadges::getBadgeFor( $this->targetLang, self::ANDROID_PLATFORM )
+				];
+			}
+			if ( $mobileApplicationsLinks[self::IOS_PLATFORM] ) {
+				$badges[self::IOS_PLATFORM] = [
+					'link' => $mobileApplicationsLinks[self::IOS_PLATFORM],
+					'src' => EmailMobileBadges::getBadgeFor( $this->targetLang, self::IOS_PLATFORM )
+				];
+			}
+		}
+
+		return $badges;
+	}
+
+	private function generateMobileApplicationsLinks() {
+		global $wgMemc;
+
+		$mobileApplicationsLinks = $wgMemc->get( $this->createCacheKeyForMobileApplicationsLinks() );
+
+		if ( is_bool( $mobileApplicationsLinks )) {
+			$mobileApplicationsLinks = [];
+			$response = $this->fetchMobileApplicationsDetails();
+
+			if ( $response && $this->applicationsExistFor( $this->wg->CityId, $response )) {
+					$mobileApplications = json_decode($response, true);
+					$mobileApplicationsLinks = $this->traverseThrough($mobileApplications);
+			}
+
+			// Event if response is not valid (for example there are some difficulties with mobile applications service)
+			// result will be cached to prevent too long delays in email generation
+			$wgMemc->set(
+				$this->createCacheKeyForMobileApplicationsLinks(),
+				$mobileApplicationsLinks,
+				time() + self::MOBILE_APPLICATIONS_LINKS_EVICTION_TIME );
+		}
+
+		return $mobileApplicationsLinks;
+	}
+
+	private function createCacheKeyForMobileApplicationsLinks() {
+		return "{$this->wg->CityId}:mobileApplicationsLinks";
+	}
+
+	/**
+	 * @return string or boolean (because of Http::request)
+	 */
+	private function fetchMobileApplicationsDetails() {
+		// currently it does not matter if Android or iOS value is added, data is returned for both Android and iOS
+		return Http::request( "GET",  'https://services.wikia.com/mobile-applications/platform/android' );
+	}
+
+
+	/**
+	 * @param string siteId - current site id
+	 * @param string response - raw, not parsed response
+	 *
+	 * @return boolean
+	 */
+	private function applicationsExistFor( $siteId, $response ) {
+		return strpos( $response, 'wikia_id":' . $siteId ) !== false;
+	}
+
+	/**
+	 * @param array $mobileApplications - array containing information about mobile applications
+	 * @return array - containing mobile applications links
+	 */
+	private function traverseThrough( $mobileApplications ) {
+		$mobileApplicationsLinks = [];
+		$siteId = $this->wg->CityId;
+
+		foreach ( $mobileApplications['apps'] as $app ) {
+			foreach ( $app['languages'] as $language ) {
+				if ( $language['wikia_id'] ==  $siteId) {
+					if ( $app['android_release'] ) {
+						$release = $app['android_release'];
+						$mobileApplicationsLinks[self::ANDROID_PLATFORM] = "https://play.google.com/store/apps/details?id=$release";
+					}
+					if ( $app['ios_release'] ) {
+						$release = $app['ios_release'];
+						$mobileApplicationsLinks[self::IOS_PLATFORM] = "https://itunes.apple.com/us/app/id$release";
+					}
+					break;
```

Comment:
I think maybe you want to `return` here? A `break` will break out of the inner loop, but the outer loop will keep going.